experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/schemaBuilder.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Assume, RestrictiveReadonlyRecord, transformObjectMap } from "../../../util";
+import { typeNameSymbol } from "../../contextuallyTyped";
+import { FieldKind } from "../../modular-schema";
+import { ImplicitFieldSchema, NormalizeField, SchemaBuilderBase } from "../../schemaBuilderBase";
+import {
+	FieldSchema,
+	InternalTypedSchemaTypes,
+	StructSchema,
+	TreeNodeSchema,
+	Unenforced,
+} from "../../typed-schema";
+import { ProxyNode, SharedTreeObject } from "./types";
+
+const factoryContent = Symbol("Node content");
+interface HasFactoryContent<T> {
+	[factoryContent]: T;
+}
+
+/**
+ * Returns the content stored on an object created by a {@link SharedTreeObjectFactory}.
+ */
+export function getFactoryContent(x: unknown): unknown | undefined {
+	return (x as Partial<HasFactoryContent<unknown>>)[factoryContent];
+}
+
+/**
+ * Creates `{@link SharedTreeObject}`s of some type via a `create` method.
+ * @alpha
+ */
+export interface SharedTreeObjectFactory<TSchema extends TreeNodeSchema<any, any>> {
```

Comment:
Don't use `any` unless you are dealing with invariant or contravariant type parameters.

This should work fine:
```suggestion
export interface SharedTreeObjectFactory<TSchema extends TreeNodeSchema> {
```

If for some reson thats way too restrictive, use "unknown" and not "any" as these paramaters are covariant and its safer to just not use any where there is no reason to use it.

Reply:
Also if the interface is just going to assume its a StructSchema, maybe the extends clause should match?

Reply:
I've tried both of those suggestions at some point and TS choked. It tends to have trouble with `StructSchema`; places where you'd think it would be fine to use `StructSchema` require crazy casts or just don't work entirely and the best I could do was use `TreeSchema` and assume. This is one thing I'm hoping that your class-based schema will improved. Still, I'll try both your suggestions again in case something has changed.